File: DP/279 Perfect Squares/solution1.py
class Solution(object):
    def numSquares(self, n):
        """
        :type n: int
        :rtype: int
        """
        
        res = n
        for i in range(2, int(math.sqrt(n)) + 1):
            a = n // (i * i)
            b = n % (i * i)
            res = min(res, a + self.numSquares(b))
        
        return res

    
if __name__ == '__main__':

    assert (Solution().numSquares(12) == 3)

    assert (Solution().numSquares(13) == 2)

File: Bit manipulation/89 Gray Code/solution1.cpp
class Solution {
public:
    vector<int> grayCode(int n) {
        // mirror
        vector<int> res = {0};
        
        for (int i = 1; i <= n; i++) {
            int size = res.size();
            for (int k = size - 1; k >= 0; k--) {
                res.push_back((1 << i - 1) | res[k]);
            }
        }
        
        return res;
    }
};

